tests/apple-exporter/components/test-class-image.php

```diff
@@ -30,6 +30,28 @@ public function filter_apple_news_image_json( $json ) {
 		return $json;
 	}
 
+	/**
+	 * Test component type ('image' or 'photo') postmeta selection.
+	 */
+	public function test_component_type() {
+		// Create test post.
+		$post_id = self::factory()->post->create(
+			[
+				// First img becomes cover, second img is output as standard json component.
+				'post_content' => '<img src="https://placeimg.com/640/480/any" alt="Example Cover" /><img src="https://placeimg.com/640/480/any" alt="Example Image" />',
```

Comment:
Discovered that the first image in an article is dynamically cherrypicked out of the article's container components and added to the header components.  That's why there are changes to both `class-image` and `class-cover` in this PR.

This test checks that both the cover image in the header and the standard image(s) in the main article container have the correct component type applied (`'image'` or `'photo'`).

I could move a the tests for the first image (cover image) to `test-class-cover` if that makes more sense.  Will need to have the same `post_content` here though so there's standard component json to test.

Reply:
I'd recommend moving the cover test to the cover test class. If you specify a featured image for a sample post it will use that instead of plucking from the content, which makes writing the test easier. See, for example: https://github.com/alleyinteractive/apple-news/blob/465151fc15eeccc293d76fdf632920846d963627/tests/apple-exporter/components/test-class-gallery.php#L126-L133

Reply:
Nice that's handy.  Makes the test more readable/intuitive too.